components/director/internal/model/capability.go

```diff
@@ -0,0 +1,154 @@
+	"regexp"
+	"strconv"
+)
+
+// Capability missing godoc
+type Capability struct {
+	ApplicationID                *string
+	ApplicationTemplateVersionID *string
+	PackageID                    *string
+	Name                         string
+	Description                  *string
+	OrdID                        *string
+	Type                         string
+	CustomType                   *string
+	LocalTenantID                *string
+	ShortDescription             *string
+	SystemInstanceAware          *bool
+	Tags                         json.RawMessage
+	Links                        json.RawMessage
+	ReleaseStatus                *string
+	Labels                       json.RawMessage
+	Visibility                   *string
+	Version                      *Version
+	ResourceHash                 *string
+	DocumentationLabels          json.RawMessage
+	CorrelationIDs               json.RawMessage
+	*BaseEntity
+}
+
+// GetType missing godoc
+func (*Capability) GetType() resource.Type {
+	return resource.Capability
+}
+
+// CapabilityInput missing godoc
+type CapabilityInput struct {
+	OrdPackageID          *string                 `json:"partOfPackage"`
+	Tenant                string                  `json:",omitempty"`
+	Name                  string                  `json:"title"`
+	Description           *string                 `json:"description"`
+	OrdID                 *string                 `json:"ordId"`
+	Type                  string                  `json:"type"`
+	CustomType            *string                 `json:"customType"`
+	LocalTenantID         *string                 `json:"localTenantId"`
+	ShortDescription      *string                 `json:"shortDescription"`
+	SystemInstanceAware   *bool                   `json:"systemInstanceAware"`
+	Tags                  json.RawMessage         `json:"tags"`
+	Links                 json.RawMessage         `json:"links"`
+	ReleaseStatus         *string                 `json:"releaseStatus"`
+	Labels                json.RawMessage         `json:"labels"`
+	Visibility            *string                 `json:"visibility"`
+	CapabilityDefinitions []*CapabilityDefinition `json:"capabilityDefinitions"`
+	DocumentationLabels   json.RawMessage         `json:"documentationLabels"`
+	CorrelationIDs        json.RawMessage         `json:"correlationIds,omitempty"`
+
+	*VersionInput `hash:"ignore"`
+}
+
+// CapabilityDefinition missing godoc
+type CapabilityDefinition struct {
+	Type           CapabilitySpecType              `json:"type"`
+	CustomType     string                          `json:"customType"`
+	MediaType      SpecFormat                      `json:"mediaType"`
+	URL            string                          `json:"url"`
+	AccessStrategy accessstrategy.AccessStrategies `json:"accessStrategies"`
+}
+
+// Validate missing godoc
+func (cd *CapabilityDefinition) Validate() error {
+	const CustomTypeRegex = "^([a-z0-9-]+(?:[.][a-z0-9-]+)*):([a-zA-Z0-9._\\-]+):v([0-9]+)$"
```

Comment:
i think this regex is wrong - there are unnecessary dashes (the ones which are not in the code block) - `^([a-z0-9`-]`+(?:[.][a-z0-9`-`]+)*):([a-zA-Z0-9._\\-]+):v([0-9]+)$`